packages/fa-icon-chooser/src/components/fa-icon-chooser/fa-icon-chooser.tsx

```diff
@@ -611,7 +611,27 @@ export class FaIconChooser {
       override = !!svgEmbedOverrideCallback();
     }
 
-    return override || this.embedSvgPrefixes.has(prefix);
+    return override || [...this.embedSvgPrefixes].length > 0;
+  }
+
+  emitIconChooserResult(iconDefinition: IconDefinition) {
+    const { prefix, iconName } = iconDefinition;
+    const iconLookup = { prefix, iconName };
+
+    const embedProSvg = get(this.kitMetadata, 'permits.embedProSvg', []);
+
+    // default to the restrictive case
+    let result = iconLookup;
+
+    const embedAllowed = this.shouldEmitSvgData() && embedProSvg.length > 0;
+
+    // upgrade it if the conditions allow. calculate `embedAllowed` using the same logic you have now.
```

Comment:
```suggestion
```

Reply:
might as well clean that up. the comment doesn't make the same sense in this context.